src/MOIWrapper.jl

```diff
@@ -1,408 +1,435 @@
     else
-        error("Either row_lower or row_upper must be of abs less than 1e20")
+        Clp.set_log_level(model.inner, 1)
     end
+    return nothing
 end
 
-function LQOI.get_linear_constraint(instance::Optimizer, row::Int)::Tuple{Vector{Int}, Vector{Float64}}
-    A = get_constraint_matrix(instance.inner)
-    A_row = A[row,:]
-    return Array{Int}(A_row.nzind), A_row.nzval
+function MOI.get(model::Optimizer, ::MOI.Silent)
+    return Clp.log_level(model.inner) == 0
 end
 
-function LQOI.change_objective_coefficient!(instance::Optimizer, col, coef)
-    objcoefs = get_obj_coefficients(instance.inner)
-    objcoefs[col] = coef
-    chg_obj_coefficients(instance.inner, objcoefs)
-end
+MOI.supports(::Optimizer, ::MOI.TimeLimitSec) = true
 
-function LQOI.change_rhs_coefficient!(instance::Optimizer, row, coef)
-    lower_bounds = get_row_lower(instance.inner)
-    upper_bounds = get_row_upper(instance.inner)
-    lower_bound = replace_inf(lower_bounds[row])
-    upper_bound = replace_inf(upper_bounds[row])
-    if lower_bound > -Inf
-        lower_bounds[row] = coef
-        chg_row_lower(instance.inner, lower_bounds)
-    elseif upper_bound < Inf
-        upper_bounds[row] = coef
-        chg_row_upper(instance.inner, upper_bounds)
+function MOI.set(model::Optimizer, ::MOI.TimeLimitSec, value)
+    if value === nothing
+        # De-activate time limit
+        Clp.set_maximum_seconds(model.inner, -1.0)
     else
-        error("Either row_lower or row_upper must be of abs less than 1e20")
+        Clp.set_maximum_seconds(model.inner, value)
     end
+    return nothing
 end
 
-function LQOI.delete_linear_constraints!(instance::Optimizer, start_row::Int, end_row::Int)
-    delete_rows(instance.inner, [Cint(i-1) for i in start_row:end_row])
-end
+# Clp behaves weird here
+MOI.get(model::Optimizer, ::MOI.TimeLimitSec) = Clp.maximum_seconds(model.inner)
+
+MOI.supports(::Optimizer, ::MOI.NumberOfThreads) = false
+
+
+# =====================
+# 
+# =======================
+MOI.supports_constraint(::Optimizer,
+    ::Type{MOI.ScalarAffineFunction{Float64}}, ::Type{S}
+) where{S <: SCALAR_SETS} = true
+
+MOI.supports_constraint(::Optimizer,
+    ::Type{MOI.SingleVariable}, ::Type{S}
+) where{S <: SCALAR_SETS} = true
+
+MOI.supports(::Optimizer, ::MOI.ObjectiveSense) = true
 
-function LQOI.change_linear_constraint_sense!(instance::Optimizer, rows::Vector{Int}, senses::Vector{Cchar})
-    lower = replace_inf.(get_row_lower(instance.inner))
-    upper = replace_inf.(get_row_upper(instance.inner))
-    for (sense, row) in zip(senses, rows)
-        lb = lower[row]
-        ub = upper[row]
-        if lb > -Inf
-            rhs = lb
-        elseif ub < Inf
-            rhs = ub
-        else
-            error("Either row_lower or row_upper must be of abs less than 1e20")
+MOI.supports(::Optimizer,
+    ::MOI.ObjectiveFunction{<:Union{
+        MOI.ScalarAffineFunction{Float64},
+        MOI.SingleVariable
```

Comment:
You don't have to support `MOI.SingleVariable`, there is a bridge for transforming it to `ScalarAffineFunction`